### PR TITLE
Add logs based alerts

### PR DESCRIPTION
**Pull request by jgbernalp**
https://issues.redhat.com/browse/OU-103
https://issues.redhat.com/browse/OU-104

This PR will add:

- Metrics component to display loki based metrics on the alert detail view
- Fetch function to get alerts from loki so console can merge with the current prometheus alerts
- Add `config/alerts.patch.json` to provide the `alerts` feature on allowed OCP versions

https://user-images.githubusercontent.com/5461414/214549412-78c87f65-a1a2-46bd-9946-3b2ef00a047f.mov

### CODE REVIEW

File: web/src/logs.types.ts [hunk 1]
```diff
@@ -55,3 +56,21 @@ export type QueryRangeResponse<T = MatrixResult | StreamsResult> = {
...
+      rules: Array<{
+        labels: Record<string, number>;
+        annotations: Record<string, number>;
+        activeAt: string;
+        state: string;
+        value: string;
+      }>;
+      annotations: Record<string, number>;
+      labels: Record<string, number>;
+      query: string;
```
Comment (kyoto) [inline web/src/logs.types.ts:73]: Is this correct? Shouldn't `annotations` and `labels` only be within the rule object or the alert object?

Similarly for `query`, which I think should be `expr` on the `rule` object?

If we are not actually using these attributes, maybe better to remove them from the type anyway.

File: web/src/components/logs-metrics.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+const CursorVoronoiContainer = createContainer('voronoi', 'cursor');
+
+export const LogsMetrics: React.FC<LogsMetricsProps> = ({
+  logsData,
+  isLoading,
+  error,
+  timeRange,
+}) => {
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+  const [width, setWidth] = React.useState<number>(0);
```
Comment (kyoto) [inline web/src/components/logs-metrics.tsx:86]: Console has a hook for handling resize: https://github.com/openshift/console/blob/master/frontend/public/components/utils/ref-width-hook.ts, which has been in use for a while so the code should be well tested.

Ideally we would expose it to the SDK, but I don't think that should hold up this PR.

File: web/src/components/alerts/logs-alerts-metrics.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  }, [rule?.query, timeRange]);
+
+  const handleTimeRangeChange = (newTimeRange: TimeRange) => {
+    setTimeRange(newTimeRange);
+  };
+
+  return (
+    <div className="co-logs-alert-metrics__container">
+      <div className="co-logs-metrics__header">
+        <TimeRangeDropdown value={timeRange} onChange={handleTimeRangeChange} />
```
Comment (kyoto) [inline web/src/components/alerts/logs-alerts-metrics.tsx:40]:
```suggestion
        <TimeRangeDropdown value={timeRange} onChange={setTimeRange} />
```

**Review by kyoto — CHANGES_REQUESTED** [review]

File: web/src/logs.types.ts (hunk 1 above)
Comment (jgbernalp) [reply web/src/logs.types.ts:73]: good catch, the type was duplicating some fields based on https://prometheus.io/docs/prometheus/latest/querying/api/#rules. I fixed it to include the labels and query which are required for the detail page to query the correct tenant.

**Review by kyoto — COMMENTED** [review]
/lgtm

**Review by kyoto — APPROVED** [review]

**Pull request merged**